app/lib/common/providers/room_providers.dart

```diff
@@ -169,32 +169,41 @@ final spaceRelationsProvider = FutureProvider.autoDispose
   return await room.spaceRelations();
 });
 
-/// Get the canonical parent of the space. Errors if the space isn't found. Stays up
+/// Get the canonical parents of the space. Errors if the space isn't found. Stays up
 /// to date with underlying client data if a space was found.
-final canonicalParentProvider = FutureProvider.autoDispose
-    .family<SpaceWithProfileData?, String>((ref, roomId) async {
+final canonicalParentsProvider = FutureProvider.autoDispose
+    .family<List<SpaceWithProfileData>, String>((ref, roomId) async {
```

Comment:
okay, naming things ... there is only one _canonical_ parent. If there are more then that's just `parentsProvider` (drop the `canonical`).